### PR TITLE
ensure not old prysm installed when installing web3signer

### PR DESCRIPTION
**Pull request by pablomendezroyo**
Web3signer should not be allow to be installed if Prysm legacy is present

### CODE REVIEW

File: packages/dappmanager/src/modules/installer/ensureEth2MigrationRequirements.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * Ensure the following requirements for Eth2 migration:
+ * - From dappmanager v0.2.46 prevent from installing legacy version (previous to remote signer support) for packages: (Prysm, Prysm-prater and Gnosis-Beacon-Chain-Prysm)
+ * -
+ */
+export async function ensureEth2MigrationRequirements(
+  packagesData: InstallPackageData[]
+): Promise<void> {
+  for (const pkg of packagesData) {
+    params.minimumAllowedPackageVersions.forEach(
```
Comment (dapplion) [inline packages/dappmanager/src/modules/installer/ensureEth2MigrationRequirements.ts:15]: forEach cannot handle async functions, the thrown errors won't stop the installation

File: packages/dappmanager/src/modules/installer/ensureEth2MigrationRequirements.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      async ({ dnpName, version }) => {
+        // Ensure is not a validator client legacy package
+        if (pkg.dnpName === dnpName && semver.lte(pkg.semVersion, version))
+          throw Error(
+            `${pkg.dnpName}:${pkg.semVersion} is a legacy validator client, install a more recent version with remote signer support`
+          );
+        // Ensure not installing web3signer if legacy client is installed
+        if (pkg.dnpName === params.web3SignerDnpName) {
```
Comment (dapplion) [inline packages/dappmanager/src/modules/installer/ensureEth2MigrationRequirements.ts:23]: You don't need to put this inside the `params.minimumAllowedPackageVersions` loop

File: packages/dappmanager/src/modules/installer/ensureEth2MigrationRequirements.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          const prysmPackage = await listPackageNoThrow({ dnpName: dnpName });
```
Comment (dapplion) [inline packages/dappmanager/src/modules/installer/ensureEth2MigrationRequirements.ts:24]: Why is dnpName the prysm package?

File: packages/dappmanager/src/modules/installer/ensureEth2MigrationRequirements.ts (hunk 1 above)
Comment (dapplion) [reply packages/dappmanager/src/modules/installer/ensureEth2MigrationRequirements.ts:15]: Convert `params.minimumAllowedPackageVersions` into a `Map<string, string>` where key is dnpName, and value is minimum version. This way you don't have a quadratic loop

**Review by dapplion — CHANGES_REQUESTED** [review]
See above

File: packages/dappmanager/src/params.ts [hunk 4]
```diff
@@ -151,16 +151,21 @@ const params = {
...
     {
       dnpName: "prysm-prater.dnp.dappnode.eth",
       version: "0.1.7"
     }
   ],
 
-  // Web3signer dnpname
-  web3SignerDnpName: "web3signer-prater.dnp.dappnode.eth",
+  // Web3signer and clients dnpnames
+  web3SignerClientsDnpNames: [
```
Comment (dapplion) [inline packages/dappmanager/src/params.ts:163]: This is really a package exclusion list. A package excludes some other packages from being installable

Comment (pablomendezroyo) [reply packages/dappmanager/src/params.ts:163]: thats correct, a package within a version range exclude the installation of the web3signer for the network of the package

**Review by dapplion — APPROVED** [review]

**Pull request merged**